services/apps/alcs/src/alcs/search/inquiry/inquiry-advanced-search.service.ts

```diff
@@ -1,31 +1,70 @@
-import { Injectable } from '@nestjs/common';
+import { RedisService } from '@app/common/redis/redis.service';
+import { Injectable, Logger } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
-import { Brackets, Repository, SelectQueryBuilder } from 'typeorm';
-import {
-  getNextDayToPacific,
-  getStartOfDayToPacific,
-} from '../../../utils/pacific-date-time-helper';
+import * as hash from 'object-hash';
+import { QueryRunner, Repository } from 'typeorm';
 import { formatStringToPostgresSearchStringArrayWithWildCard } from '../../../utils/search-helper';
+import { intersectSets } from '../../../utils/set-helper';
 import { InquiryParcel } from '../../inquiry/inquiry-parcel/inquiry-parcel.entity';
+import { Inquiry } from '../../inquiry/inquiry.entity';
 import { LocalGovernment } from '../../local-government/local-government.entity';
 import { AdvancedSearchResultDto, SearchRequestDto } from '../search.dto';
 import { InquirySearchView } from './inquiry-search-view.entity';
 
 @Injectable()
 export class InquiryAdvancedSearchService {
+  private logger: Logger = new Logger(InquiryAdvancedSearchService.name);
+
   constructor(
     @InjectRepository(InquirySearchView)
     private inquirySearchViewRepository: Repository<InquirySearchView>,
+    @InjectRepository(Inquiry)
+    private inquiryRepository: Repository<Inquiry>,
     @InjectRepository(LocalGovernment)
     private governmentRepository: Repository<LocalGovernment>,
+    private redisService: RedisService,
   ) {}
 
   async search(
     searchDto: SearchRequestDto,
+    queryRunner: QueryRunner,
   ): Promise<AdvancedSearchResultDto<InquirySearchView[]>> {
-    let query = await this.compileInquirySearchQuery(searchDto);
+    const searchHash = hash(searchDto);
+    const searchKey = `search_alcs_inquiry_${searchHash}`;
+
+    const client = this.redisService.getClient();
+    const cachedSearch = await client.get(searchKey);
+
+    let fileNumbers = new Set<string>();
+    if (cachedSearch) {
+      const cachedNumbers = JSON.parse(cachedSearch) as string[];
+      fileNumbers = new Set<string>(cachedNumbers);
+    } else {
+      fileNumbers = await this.searchForFileNumbers(searchDto);
+      await client.setEx(
+        searchKey,
+        180, //Seconds
```

Comment:
nitpick: this expiration is shared across all searches would be nice to have it in constant

Reply:
Done! Moved to search.config.ts but may move again after refactor